# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1387314155 21600
#      Tue Dec 17 15:02:35 2013 -0600
# Branch gui-release
# Node ID f8b7e8483b88c1ef14bd0bfc399a833a57d28b12
# Parent  7721e78b1337a292a9900326d1313fe2d0f1d302
avoid possible num-to-str warnings from strread.m

* strread.m: Use "\n" instead of 10 and " " instead of 32 in
comparisions with character strings.

diff --git a/scripts/io/strread.m b/scripts/io/strread.m
--- a/scripts/io/strread.m
+++ b/scripts/io/strread.m
@@ -391,25 +391,25 @@ function varargout = strread (str, forma
     white_spaces = strrep (white_spaces, eol_char, '');
   endif
 
   pad_out = 0;
   ## Trim whitespace if needed
   if (! isempty (white_spaces))
     ## Check if trailing "\n" might signal padding output arrays to equal size
     ## before it is trimmed away below
-    if ((str(end) == 10) && (nargout > 1))
+    if (str(end) == "\n" && nargout > 1)
       pad_out = 1;
     endif
     ## Condense all repeated whitespace into one single space
     ## FIXME: this will also fold repeated whitespace in a char field
     rxp_wsp = sprintf ("[%s]+", white_spaces);
     str = regexprep (str, rxp_wsp, ' ');
     ## Remove possible leading space at string
-    if (str(1) == 32)
+    if (str(1) == " ")
        str = str(2:end);
     endif
     ## Check for single delimiter followed/preceded by whitespace
     if (! isempty (delimiter_str))
       dlmstr = setdiff (delimiter_str, " ");
       if (! isempty (dlmstr))
         rxp_dlmwsp = sprintf ('( [%s] | [%s]|[%s] )', dlmstr, dlmstr, dlmstr);
         str = regexprep (str, rxp_dlmwsp, delimiter_str(1));
